pkg/services/k8s/client/clientset.go

```diff
@@ -102,12 +103,12 @@ type ShortWebhookConfig struct {
 // ProvideClientset returns a new Clientset configured with cfg.
 func ProvideClientsetProvider(toggles featuremgmt.FeatureToggles, restConfigProvider apiserver.RestConfigProvider, cfg *setting.Cfg) (*service, error) {
 	s := &service{restConfigProvider: restConfigProvider, grafanaCfg: cfg}
-	s.BasicService = services.NewBasicService(s.start, s.running, nil)
-
+	s.BasicService = services.NewBasicService(s.start, s.running, nil).WithName(modules.KubernetesClientset)
 	return s, nil
 }
 
 func (s *service) GetClientset() *Clientset {
+	_ = s.AwaitRunning(context.Background())
```

Comment:
calling `AwaitRunning` here should prevent returning `nil` if something calls `GetClientset` before the clientset is available